engines/native/sandbox.go

```diff
@@ -32,10 +32,11 @@ type sandbox struct {
 	shells        []*shell
 }
 
-func newSandbox(b *sandboxBuilder) (s *sandbox, err error) {
+func newSandbox(b *sandboxBuilder) (engines.Sandbox, error) {
```

Comment:
This change was needed because we do `return newSandbox(...)` in `SandboxBuilder` and when `newSandbox` returns `*sandbox(nil)` it gets converted to `engines.Sandbox != nil`, I know that messed up.. But `nil` is a valid value for `*sandbox` and `*sandbox` implements `engines.Sandbox`... So if we're not careful to explicitly return nil, we return a nil implementation... which cases issues if you call methods on it..